app/src/lib/helpers/linux.ts

```
@@ -0,0 +1,85 @@
+import { join } from 'path'
+import { pathExists as pathExistsInternal } from 'fs-extra'
+import {
+  ChildProcess,
+  spawn,
+  SpawnOptionsWithoutStdio,
+  SpawnOptions,
+} from 'child_process'
+
+export function isFlatpakBuild() {
+  return __LINUX__ && process.env.FLATPAK_HOST === '1'
+}
+
+/**
+ * Convert an executable path to be relative to the flatpak host
+ *
+ * @param path a path to an executable relative to the root of the filesystem
+ *
+ */
+function convertToFlatpakPath(path: string) {
+  return join('/var/run/host', path)
+}
+
+/**
+ * Checks the file path on disk exists before attempting to launch a specific shell
+ *
+ * @param path
+ *
+ * @returns `true` if the path can be resolved, or `false` otherwise
+ */
+export async function pathExists(path: string): Promise<boolean> {
+  if (isFlatpakBuild()) {
+    path = convertToFlatpakPath(path)
```

🤔 as I understand it, we want to convert program paths before checking if we're running in a flatpak context as they exist in a special location, but we don't need to do this when we want to launch the program - `flatpak-spawn` will do that itself as long as we call it with `--host`.

Let me know if this logic is incorrect, or if I've misunderstood.

---

yup this is correct. the path outside the container should be real but detection for binaries in `/usr` should be pointed to `/var/run/host`

---

What about for other special directories we're using for editors:

 - `/opt`
 - `/snap` 
 - `/var/lib/flatpak` (launching another flatpak program)

Should those also go through this conversion process before we try and open them with `flatpak-spawn`?

---

I think `/var/lib/flatpak` might be blocked by flatpak not sure(I'll check and update this), `/snap`  is one that won't work since this dir won't be mounted

---

yeah `/var/lib/flatpak` is blocked

---

👍🏻 I'm assuming that the `try-catch` below will ensure these errors will be handled and swallowed by the app, so I'm going to try and avoid getting too complex with the logic in here for now for paths that we know are inaccessible in the flatpak context.